boa_engine/src/bytecompiler/mod.rs

```diff
@@ -544,15 +544,16 @@ impl<'b, 'host> ByteCompiler<'b, 'host> {
     }
 
     fn patch_jump_with_target(&mut self, label: Label, target: u32) {
+        const U32_SIZE: usize = std::mem::size_of::<u32>();
+
         let Label { index } = label;
 
         let index = index as usize;
-
         let bytes = target.to_ne_bytes();
-        self.bytecode[index + 1] = bytes[0];
-        self.bytecode[index + 2] = bytes[1];
-        self.bytecode[index + 3] = bytes[2];
-        self.bytecode[index + 4] = bytes[3];
+
+        // This is done to avoid unneeded bounds checks.
+        assert!(self.bytecode.len() > index + U32_SIZE && usize::MAX - U32_SIZE >= index);
+        self.bytecode[index..index + U32_SIZE].copy_from_slice(bytes.as_slice());
```

Comment:
Wouldn't we want index + 1 here?

Reply:
Yup, its offset by one, that's why all the tests are failing :sweat_smile:

Reply:
Oops, I directly optimized https://godbolt.org/z/q15WGh9n8 without considering the original context from where it was taken 😅

Reply:
Should be
```javascript
const U32_SIZE: usize = std::mem::size_of::<u32>();
assert!(code.len() > index + U32_SIZE && usize::MAX - U32_SIZE >= index);

let values = value.to_ne_bytes();
code[index + 1..=index + U32_SIZE].copy_from_slice(values.as_slice())
```
then

Reply:
Would it be more clear to put something like:

```rust
let slice_index = index + 1;
code[slice_index..slice_index + U32_SIZE].copy_from_slice(values.as_slice())
```

Adding the extra line may not be necessary, but it may help with readability down the line.

Reply:
Maybe `U32_SIZE` is making the snippet a bit less readable. Something like
```Rust
// u32 (4 bytes in size) should fit inside `code`
assert!(code.len() > index + 4 && usize::MAX - 4 >= index);

let values = value.to_ne_bytes();
code[index + 1..index + 4].copy_from_slice(values.as_slice());
```
Should be fine

Reply:
> Would it be more clear to put something like:
> 
> ```rust
> let slice_index = index + 1;
> code[slice_index..slice_index + U32_SIZE].copy_from_slice(values.as_slice())
> ```
> 
> Adding the extra line may not be necessary, but it may help with readability down the line.

This doesn't optimize the indexing :(
https://godbolt.org/z/hzbhW7e3c

Reply:
The previous one with the constant is fine IMO.